smaug/python/tensor_utils.py

```diff
@@ -0,0 +1,68 @@
+from smaug.core import types_pb2
+from smaug.python.tensor import Tensor
+
+def get_tensor_data(tensor_data_array, tensor_name):
+  """Find the tensor data for this tensor by its name."""
+  for i in range(len(tensor_data_array.data_array)):
+    if tensor_data_array.data_array[i].name == tensor_name:
+      return tensor_data_array.data_array[i]
+  return None
+
+def account_for_padding(shape):
+  """Return a TensorShapeProto that has its dims padded."""
+  alignment = shape.alignment
+  remainder = shape.dims[-1] % shape.alignment
+  if alignment == 0 or remainder == 0:
+    return shape
+  shape.dims[-1] += alignment - remainder;
+  return shape
+
+def from_tensor_proto(tensor_proto, tensor_data_array=None):
```

Comment:
This is because `targets` are `NodeProto`s, and in a `NodeProto` we can only get a `TensorProto`, whereas we want is a `Tensor`. Yes the goal is to get an existing DataOp and return its output tensor (i.e. a `Tensor`).

Reply:
Then I think the better way to deal with this issue is to just keep a reference to the output tensor directly. With `from_tensor_proto`, you will end up creating a new instance of a Tensor that is otherwise identical to an existing one, which is not only inefficient, it's almost guaranteed to cause confusion later down the road (tensor0 != tensor1 unless you implement `__eq__` and compare *every single one of its elements*).

To clarify, what I mean is, for every tensor that is an input to a data op, store a reference to the output of the data op when the data op is created in the original input tensor.